src/main.rs

```diff
@@ -52,6 +57,20 @@ fn open_dit_repo() -> Result<Repository> {
 }
 
 
+/// Get a vector of commits from values
+///
+/// This function transforms values to a vector.
+///
+fn values_to_hashes<'repo>(repo: &'repo Repository, values: Values) -> Result<Vec<Commit<'repo>>> {
+    let mut retval = Vec::new();
+    for commit in values.map(|string| repo.revparse_single(string))
+                        .map(|oid| repo.find_commit(try!(oid).id())) {
+        retval.push(try!(commit));
+    }
+    Ok(retval)
+}
```

Comment:
It is not really nice that you collect the iterator here and start a new iterator from the `Vec` below, but I let that slip because I don't see any value in optimizing this right now and we can improve it later anyways.